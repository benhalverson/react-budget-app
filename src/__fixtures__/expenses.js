import moment from 'moment';

export default [
  {
    id: '1',
    description: 'Gum',
    notes: '',
    amount: 195,
    createdAt: 0
  },
  {
    id: '2',
    description: 'Rent',
    notes: '',
    amount: 1234,
    createdAt: moment(0).subtract(4, 'days').valueOf()
  },
  {
    id: '3',
    description: 'Credit Card',
    notes: '',
    amount: 5678,
    createdAt: moment(0).add(4, 'days').valueOf()
  }
];
