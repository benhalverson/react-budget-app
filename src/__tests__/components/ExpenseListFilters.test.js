import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import configureStore from 'redux-mock-store';
import ConnectedExpenseListFilters, {ExpenseListFilters} from '../../components/ExpenseListFilters';
import { filters, altFilters } from '../../__fixtures__/filters';


let setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  sortByDate = jest.fn();
  sortByAmount = jest.fn();
  setStartDate = jest.fn();
  setEndDate = jest.fn();
  wrapper = shallow(<ExpenseListFilters
    filters={filters}
    setTextFilter={setTextFilter}
    sortByDate={sortByDate}
    sortByAmount={sortByAmount}
    setStartDate={setStartDate}
    setEndDate={setEndDate}
    />)
});

test('should render ExpenseListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseListFilters with alt data correctly', () => {
  wrapper.setProps({
    filters: altFilters
  });
  expect(wrapper).toMatchSnapshot();
});

test('should handle text change', () => {
  const value = 'rent';
  wrapper.find('input').simulate('change', {
    target: {
      value
    }
  });
  expect(setTextFilter).toHaveBeenLastCalledWith(value)
});

test('should sort by date', () => {
  const value = 'date';
  wrapper.setProps({
    filters: altFilters
  });
  wrapper.find('select').simulate('change', {
    target: { value }
  });
  expect(sortByDate).toHaveBeenCalled();
});

test('should sort by amount', () => {
  const value = 'amount';
  wrapper.find('select').simulate('change', {
    target: { value }
  });
  expect(sortByAmount).toHaveBeenCalled();
});

test('should handle date changes', () => {
  const startDate = moment(0).add(4, 'years');
  const endDate = moment(0).add(8, 'years');
  wrapper.find('withStyles(DateRangePicker)').prop("onDatesChange")({ startDate, endDate });
  expect(setStartDate).toHaveBeenLastCalledWith(startDate);
  expect(setEndDate).toHaveBeenLastCalledWith(endDate);
}) 

test('should handle date focus changes', () => {
  const calendarFocused = 'endDate';
  wrapper.find('withStyles(DateRangePicker)').prop('onFocusChange')(calendarFocused);
  expect(wrapper.state('calendarFocused')).toBe(calendarFocused);
});

test('should render ExpenseListFilters with props from redux', () => {
  const mockStore = configureStore();
  const store = mockStore({ filters })
  const wrapper = shallow(<ConnectedExpenseListFilters store={store} />);

  expect(wrapper.props().filters).toEqual(filters);
});