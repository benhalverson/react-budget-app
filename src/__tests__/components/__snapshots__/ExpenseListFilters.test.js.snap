// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListFilters correctly 1`] = `ShallowWrapper {}`;

exports[`should render ExpenseListFilters with alt data correctly 1`] = `ShallowWrapper {}`;
